### DISCUSSION TITLE
Drone Model, high overlap.

### DISCUSSION THREAD

**Discussion by aguirrejuan** [weecology/DeepForest#548] [Q&A]
**Please describe a feature you would like to be added.**
I find it challenging to accurately detect and classify different species of trees in various landscapes using current image recognition software. 

**Describe the solution you'd like**
I would like an integration with YOLOv8 from Ultralytics, which is state-of-the-art in object detection. This could enhance the ability to discern between tree species by leveraging YOLOv8's high precision and real-time processing capabilities.

**Comment by bw4sz** [comment]
Can you show some data and model training that fails given the current retinanet backbone? We have always believed that data, not architecture, is the limiting factor here. No doubt that the new YOLO models have great functionality and speed, but given the data sizes the vast majority of ecological projects, I have yet to see an example in which the two backbones, given the same data, produce meaningfully different, let's say 4% results. I am 100% open to including new models, and we hope to make the framework open to any particular kind of model a user wants to bring in. Having examples in which the current workflow is insufficient, but another model is better, would help immensely in speeding up this work.

As an aside, we are retraining the detection backbone currently, are you able to share annotations and or images so that it can be improved for your use case? See https://github.com/weecology/DeepForest/issues/340

**Comment by aguirrejuan** [comment]
Hello, thank you for your prompt reply.

I'm encountering difficulties with the following types of images:

![annotations_58](https://github.com/weecology/DeepForest/assets/50622777/fff8b51b-3780-4022-a18d-8854af106bec)
![annotations_57](https://github.com/weecology/DeepForest/assets/50622777/7f2cdc0c-487f-41bf-8bcb-fd00f179a697)
![annotations_56](https://github.com/weecology/DeepForest/assets/50622777/5c42dadd-166b-4071-a847-55db919fc593)
![annotations_55](https://github.com/weecology/DeepForest/assets/50622777/0288828b-59f5-479f-b3ac-28fc31723179)

I'm using a total of 169 images for training. To address the issue of high overlap, I've increased the nms_thresh parameter, which resulted in the following metrics:

Box Precision: 0.9
Box Recall: 0.55
My goal is to improve the recall further. Would you have any suggestions on how I might achieve that?

Thanks

**Comment by bw4sz** [comment]
Great, I've made this into a discussion. Can you drop any code you've used and I can look through it. If you can share the data I will have a quick run and see if I can recreate your score. I'm sure we can do better than that recall, I'll know more looking at the code, but that precision score is quite high compared to recall. The images look preprocessed in some interesting way, how were they acquired? Can you tell me about the project?

**Reply by aguirrejuan** [reply]
Hi, thanks for the response,  

My code is in this [repo](https://github.com/aguirrejuan/CountTrees), and I am running it using the next [colab](https://colab.research.google.com/drive/1QGSQ0Nt34Lca1ZARSD5XR41koFI4pUnL?usp=sharing). 

Unfortunately, I cannot share with you the raster itself (even though I have shared the png images in the colab), but I can give you some metadata about it:

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /></head><body>


Extent | 613125.3492000000551343,6552856.3001172002404928 : 614684.2772000000113621,6554128.8089000005275011
-- | --
Width | 8186
Height | 6682
Data type | Float32 - Thirty two bit floating point
GDAL Driver Description | GTiff
GDAL Driver Metadata | GeoTIFF
Compression |  
Band 1 | RepresentationType=ATHEMATIC STATISTICS_COVARIANCES=2.473962004076284e-06,6.882629207049276e-06,2.721491243990084e-06 STATISTICS_MAXIMUM=0.034067593514919 STATISTICS_MEAN=0.0034678940770876 STATISTICS_MINIMUM=0.00030788040021434 STATISTICS_SKIPFACTORX=1 STATISTICS_SKIPFACTORY=1 STATISTICS_STDDEV=0.0015728833408986 Scale: 1 Offset: 0
Band 2 | RepresentationType=ATHEMATIC STATISTICS_COVARIANCES=6.882629207049276e-06,2.522084463718717e-05,8.7630580751556e-06 STATISTICS_MAXIMUM=0.049846604466438 STATISTICS_MEAN=0.0091539731403973 STATISTICS_MINIMUM=0.0010783111210912 STATISTICS_SKIPFACTORX=1 STATISTICS_SKIPFACTORY=1 STATISTICS_STDDEV=0.0050220359056051 Scale: 1 Offset: 0
Band 3 | RepresentationType=ATHEMATIC STATISTICS_COVARIANCES=2.721491243990084e-06,8.7630580751556e-06,3.366556095097584e-06 STATISTICS_MAXIMUM=0.026251189410686 STATISTICS_MEAN=0.0050266261296721 STATISTICS_MINIMUM=0.0011522535933182 STATISTICS_SKIPFACTORX=1 STATISTICS_SKIPFACTORY=1 STATISTICS_STDDEV=0.0018348177280312 Scale: 1 Offset: 0
More information | AREA_OR_POINT=Area DataType=Generic
Dimensions | X: 8186 Y: 6682 Bands: 3
Origin | 613125.3492000000551343,6554128.8089000005275011
Pixel Size | 0.1904383093085702805,-0.1904383093086332024

</body></html>

About the preprocessing, I do the following [preprocessing](https://github.com/aguirrejuan/CountTrees/blob/672dfbd616d805ded39ff7de854871d33a3a1589/count_trees/utils/processing_data.py#L51).
